src/client/pages/OfferNew/Introduction/Sidebar/StartDate.tsx

```diff
@@ -313,10 +262,20 @@ const DateForm: React.FC<{
         size={size}
       >
         <Value>
-          <DateLabel hasStartDate={Boolean(dateValue)}>
-            {getDateLabel()}
-          </DateLabel>
-          {hasStartDate && <DownArrow />}
+          {!hasStartDate &&
+            !isLoadingPickedStartDate &&
+            hasCurrentInsurer(quote) && (
+              <StartDateLabelSwitcher
+                dataCollectionId={quote.dataCollectionId}
+              />
+            )}
+          {isLoadingPickedStartDate && (
+            <LoadingDotsWrapper>
+              <LoadingDots color={colorsV3.gray500} />
+            </LoadingDotsWrapper>
+          )}
+          {hasStartDate && !isLoadingPickedStartDate && getDateLabel()}
+          {!isLoadingPickedStartDate && <DownArrow />}
```

Comment:
What about checking for `isLoadingPickedStartDate` so we don't have to check that in all the other statements? :) 
```suggestion
          {isLoadingPickedStartDate ? (
            <LoadingDotsWrapper>
              <LoadingDots color={colorsV3.gray500} />
            </LoadingDotsWrapper>
          ) : (
            <>
              {!hasStartDate && hasCurrentInsurer(quote) && (
                <StartDateLabelSwitcher
                  dataCollectionId={quote.dataCollectionId}
                />
              )}

              {hasStartDate && getDateLabel()}
              <DownArrow />
            </>
          )}
```

Reply:
See previous comment 👆

Reply:
But with all that said - this can definitely be written in a slightly more elegant way 😄